### ISSUE TITLE
nil child of an element stops subsequent children from being created

### ISSUE THREAD

**Issue by jjttjj** [hoplon/hoplon#159]
If there is a nil child of an element, the elements after it will not be created in the dom:
```
(body
   (h1 "Hello, Hoplon!")
   (p "foo")
   nil
   (p "bar"))
```
the `(p "bar")` is never created here.

**Comment by flyboarder** [comment]
@jjttjj what happens if the children are all in a vector? This could be an issue with converting a non-list of children into a proper list of children.

**Comment by jjttjj** [comment]
Yes it works fine if the elements in the code I posted are all in a vector. Personally I'd accept putting them in a vector but I think the issue is that it is unexpected to stop evaluating things after a nil value.

Could you explain a little more what you mean by the "issue converting a non-list of children to a proper list of children"? 

The (tiny) PR I submitted seems to handle most hoplon element I could think up for a test file, and work with a couple projects I tried it out on, but I definitely could be missing something.

**Comment by flyboarder** [comment]
The snippet you provided above is not technically correct since children should be in a list, when they are not in a list they are converted to a list which is where I imagine the problem lies. It's part of the argument parsing to handle attributes.

**Comment by flyboarder** [comment]
I agree your PR should solve this case tho!

**Comment by alandipert** [comment]
Will probably be resolved by #170
